client/src/components/UserHeader/UserHeader.tsx

```diff
@@ -2,21 +2,21 @@ import React, { useEffect, useState } from "react";
 import { User } from "src/interfaces";
 import { DEFAULT_USER_HEADER_SUBTITLE } from "src/constants";
 import { Typography, Paper, Grid, Box, Button } from "@material-ui/core";
-import { useTheme } from "@material-ui/core/styles";
 import { getDayPeriod } from "src/utils";
 import { isDebug } from "src/config";
+import Header from "src/components/Header";
 
 function UserHeader({ user }: { user: User | null }) {
   const curHour = new Date().getHours();
   const curMin = new Date().getMinutes();
   const [curSec, setCurSec] = useState();
 
-  const theme = useTheme();
-  const headerTopPadding = theme.spacing(2);
-  const headerBotPadding = theme.spacing(4);
+  let componentIsMounted = false;
```

Comment:
can consider using a react class component here if you need componentIsMounted etc, I feel its easier to read, but only a preference

Reply:
mm yup, still an issue for our project #20, currently all our components are functional components and as this is a small UI-only component, I'd say this is good enough for now (also this time feature may be removed, then the mounted check won't be necessary)